flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Expressions.java

```diff
@@ -535,6 +535,18 @@ public static ApiExpression ifThenElse(Object condition, Object ifTrue, Object i
         return apiCall(BuiltInFunctionDefinitions.IF, condition, ifTrue, ifFalse);
     }
 
+    /**
+     * Coalesce specifies a series of expressions, and returns the first expression whose value is
+     * not null. If all the expressions evaluate as null, coalesce returns a null value.
+     *
+     * <p>e.g. coalesce($("f0"), "-") leads to "-" if f0 is null
+     *
+     * @param args the input expressions.
+     */
+    public static ApiExpression coalesce(Object... args) {
```

Comment:
We'll also need to add this for the Scala API (ImplicitExpressionConversions).

Reply:
Added the python bindings as well